### PR TITLE
[Teams] Add ComposeMessagingExtensionAuthBot

### PR DESCRIPTION
**Pull request by EricDahlvang**
Fixes: https://github.com/microsoft/botbuilder-dotnet/issues/2633

### CODE REVIEW

**Comment by fuselabs** [comment]
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Builder/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.AI.Luis.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Builder.AI.Luis/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.AI.QnA.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Builder.AI.QnA/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.ApplicationInsights.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Builder.ApplicationInsights/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.Azure.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Builder.Azure/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.Dialogs.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Builder.Dialogs/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.TemplateManager.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Builder.TemplateManager/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Configuration.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Configuration/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Connector.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Connector/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Schema.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Schema/4.3.1)

File: tests/Teams/ComposeMessagingExtensionAuth/Bots/ComposeMessagingExtensionAuthBot.cs [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+                }
+            }
+        }
+
+        protected override async Task<MessagingExtensionActionResponse> OnTeamsMessagingExtensionFetchTaskAsync(ITurnContext<IInvokeActivity> turnContext, MessagingExtensionQuery query, CancellationToken cancellationToken)
+        {
+            var reply = MessageFactory.Text("OnTeamsMessagingExtensionFetchTaskAsync MessagingExtensionQuery: " + JsonConvert.SerializeObject(query));
+            await turnContext.SendActivityAsync(reply, cancellationToken).ConfigureAwait(false);
+
+            var tokenStatus = await (turnContext.Adapter as IUserTokenProvider).GetTokenStatusAsync(turnContext, turnContext.Activity.From.Id, _connectionName, cancellationToken: cancellationToken).ConfigureAwait(false);
```
Comment (johnataylor) [inline tests/Teams/ComposeMessagingExtensionAuth/Bots/ComposeMessagingExtensionAuthBot.cs]: nit: we don't usually put .ConfigureAwait(false)

File: tests/Teams/ComposeMessagingExtensionAuth/Bots/ComposeMessagingExtensionAuthBot.cs [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+
+                // Retrieve the OAuth Sign in Link to use in the MessagingExtensionResult Suggested Actions
+                var signInLink = await (turnContext.Adapter as IUserTokenProvider).GetOauthSignInLinkAsync(turnContext, _connectionName, cancellationToken).ConfigureAwait(false);
+
+                return new MessagingExtensionActionResponse
+                {
+                    ComposeExtension = new MessagingExtensionResult
+                    {
+                        Type = "auth",
+                        SuggestedActions = new MessagingExtensionSuggestedAction()
```
Comment (johnataylor) [inline tests/Teams/ComposeMessagingExtensionAuth/Bots/ComposeMessagingExtensionAuthBot.cs]: nit: you don't need () in this context

File: tests/Teams/ComposeMessagingExtensionAuth/Bots/ComposeMessagingExtensionAuthBot.cs (hunk 1 above)
Comment (EricDahlvang) [reply tests/Teams/ComposeMessagingExtensionAuth/Bots/ComposeMessagingExtensionAuthBot.cs]: Removed .ConfigureAwaits

File: tests/Teams/ComposeMessagingExtensionAuth/Bots/ComposeMessagingExtensionAuthBot.cs (hunk 2 above)
Comment (EricDahlvang) [reply tests/Teams/ComposeMessagingExtensionAuth/Bots/ComposeMessagingExtensionAuthBot.cs]: Removed (thanks!)

**Comment by fuselabs** [comment]
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Builder/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.AI.Luis.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Builder.AI.Luis/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.AI.QnA.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Builder.AI.QnA/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.ApplicationInsights.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Builder.ApplicationInsights/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.Azure.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Builder.Azure/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.Dialogs.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Builder.Dialogs/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.TemplateManager.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Builder.TemplateManager/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Configuration.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Configuration/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Connector.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Connector/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Schema.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Schema/4.3.1)

**Pull request merged**